homeassistant/components/websocket_api/commands.py

```diff
@@ -202,3 +204,40 @@ def handle_ping(hass, connection, msg):
     Async friendly.
     """
     connection.send_message(pong_message(msg["id"]))
+
+
+@callback
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "render_template",
+        vol.Required("template"): cv.template,
+        vol.Optional("entity_ids"): cv.entity_ids,
+    }
+)
+def handle_render_template(hass, connection, msg):
+    """Handle render_template command.
+
+    Async friendly.
+    """
+    template = msg["template"]
+    if template is not None:
```

Comment:
template cannot be None ?

Reply:
I was thinking this would be a fast track if there was no template that actually needed rendering... something short-circuited in my brain, I think...